source/features/linkify-action.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'dom-chef';
+
+import * as pageDetect from 'github-url-detection';
+
+import {ArrowLeftIcon} from '@primer/octicons-react';
+
+import features from '../feature-manager.js';
+import api from '../github-helpers/api.js';
+import observe from '../helpers/selector-observer.js';
+
+async function getPullRequestNumber(user: string, repository: string, runDatabaseId: string): Promise<number | false> {
+	const {resource} = await api.v4(`
+		query getPullRequestNumber($resource: URI!) {
+			resource(url: $resource) {
+				... on WorkflowRun {
+					event
+					checkSuite {
+						matchingPullRequests(last: 1) {
+							edges {
+								node {
+									number
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`, {
+		variables: {
+			resource: `/${user}/${repository}/actions/runs/${runDatabaseId}`,
+		},
+	});
+
+	if (resource.event !== 'pull_request') {
+		return false;
+	}
+
+	const {edges} = resource.checkSuite.matchingPullRequests;
+	if (edges.length === 0) {
+		return false;
+	}
+
+	return edges[0].node.number;
+}
+
+async function add(contextBar: HTMLDivElement): Promise<void> {
+	const url = new URL(location.href);
+	if (typeof url.searchParams.get('pr') === 'string') {
+		return;
+	}
+
+	const paths = url.pathname.split('/').slice(1);
+	const prNumber = await getPullRequestNumber(paths[0], paths[1], paths[4]);
+
+	if (!prNumber) {
+		return;
+	}
+
+	contextBar.classList.add('flex-column', 'flex-items-start');
+	contextBar.append(
+		<span className="PageHeader-parentLink">
```

Comment:
Instead of adding a new line, we should add it next to an existing text. The current setup pushes the whole page down when loaded, which we should avoid. See this issue I recently fixed:

- #6677 

A possible position is either next to "ci" or next to the run number:

```
{title………} #123 (PR #6666)
```

Reply:
I see that this is the native position and style in:


https://github.com/refined-github/refined-github/actions/runs/5104153489/jobs/9174884558?pr=6694

So now I'm thinking… why don't we change the links in https://github.com/refined-github/refined-github/actions to include this `pr` parameter instead of using the API? This would make the feature simpler (I'm not sure if there are other situations where this "fix" is possible)

Reply:
I did not added search params despite it is simple since the original back link button will be disappeared.

And how can we observe that? Adding some observing feature on local memory or storage? How about we visit the action page directly not caching actions page? (also pagination)

Reply:
> the original back link button will be disappeared

I don't find that button useful and GitHub agrees since they're hiding it. We can discuss this before making a decision though.

cc @kidonng too

Reply:
I agree original link is not useful

Reply:
> And how can we observe that?

The feature I'm suggesting would be similar to `hide-user-forks`. In short:

```js
observe('that pr link I mentioned', prLink => {
  prLink.closest('.row').querySelector('a').search = '?pr=' + prLink.textContent
})
```

Reply:
Oops my bad english skill.
I understood observe triggered PR number and Add back link with that information on action run page. But you said actions (list) page.
That feature is easy. What should the feature name be

Reply:
Yes, I’m suggesting that those 3 lines are the entire feature, no API calls at all.

Reply:
@fregante When we append search params, the page is being refreshed (and flickering a time). I don't think this is refined.
How about just replacing original link to new link?

Reply:
It seems cool